### DISCUSSION TITLE
Send Later

### DISCUSSION THREAD

**Discussion by reido2012** [neomutt/neomutt#4207] [Development]
Hi, 
 
I've been using Neomutt as my primary email client for a couple months now and I would like to contribute. 

I came from an email client that had a very handy feature called 'Send Later'. Most email clients have this feature now so I thought it would be a great addition. 

This feature allows a user to schedule an email to be sent at a later date. 

### How I think it could work

On the email compose/confirmation page you have the option to 'Send Later'. 

When you select 'Send Later' there are two options for getting the time and date to send the email:

1.  Allow natural language input of a date and time that we parse using a library like [dateparser](https://github.com/scrapinghub/dateparser/) (if there's a similar one in C)

2. Provide a set of options at fixed time
	- If it's currently morning
		- This afternoon
		- This evening
		- Tomorrow morning
	- If it's currently afternoon
		- This evening
		- Tomorrow morning
		- Tomorrow afernoon
	- If it's currently evening
		- Tomorrow morning
		- Tomorrow afternoon
		- Tomorrow evening

When the time is confirmed by the user we:
1.  Tag the email with time it should be sent `X-Schedule:<datetime>`
2. We move the email to a dedicated mailbox/folder called "Scheduled"

We have a background process that scans the "Scheduled" folder 

This process will move an email to the outbox to be sent as normal when the time on the email tag is the same or less than the current date time.

This is my first trying to contribute to an open source project so I'm very open to suggestions. 

 I'm curious to hear if other people are also interested in this feature. 

Thanks

**Comment by ossilator** [comment]
i think this proposal suffers from two major issues:
1. what is the use case? other than "most email clients have this feature", no motivation is provided.
2. the suggested implementation seems perfectly reasonable, but given neomutt's target audience, why would this need to be a built-in? a cron job would seem adequate.

**Comment by jindraj** [comment]
There's a way to implement send later feature on top of msmtp with msmtpq https://unix.stackexchange.com/a/554369/85046.

**Comment by ananthu-sk** [comment]
A hacky way to do is adding the below to a `cronjob` or to the `at` command

```cat '/home/username/.local/share/mail/email@uni-weimar.de/Drafts/cur/1767350757.R13198362930916393139.arch,U=21:2,S' | msmtp -t -a 'email@uni-weimar.de'```

where `/home/username/.local/share/mail/email@uni-weimar.de/Drafts/cur/1767350757.R13198362930916393139.arch,U=21:2,S` is the path of the draft and `email@uni-weimar.de` mail id from which to send.

**Comment by pinecat** [comment]
I've been using neomutt for my personal mail for a while now, and this was not something I needed. However, I recently switched to neomutt for my work mail, as well, and found I wanted a "Send Later" feature, since I work late sometimes, and don't want to bother my co-workers with emails late at night. So, I decided to build my own program to do this, which is sendmail agnostic (i.e. should work with postfix, msmtp, opensmtpd, really anything). It's a small rust program (really, suite of rust programs) which wraps [at](https://linux.die.net/man/1/at). 

It's called `midnight`, and you can install it from https://crates.io/crates/midnight, via:

```sh
cargo install midnight
```

I think the README should cover you for getting setup, but if you have any questions, feel free to ask below (assuming that is ok with the neomutt devs on this thread). The program is still a work-in-progress, and I'd like to add some more configuration options, and CLI flags for different features, but so far, I believe it is pretty usable at the moment. Because this is a wrapper for `at` and `neomutt`, rather than a specific smtp client implementation, it also has the benefit of automatically moving the sent message from your draftbox to your sent items (does msmtp do this? I do not know, sorry). If anyone tries this out and runs into a bug or has any feedback, please get in touch. Thanks!

**Reply by flatcap** [reply]
Nice!

> assuming that is ok with the neomutt devs on this thread

Ooh...  Give us a star :star: and I'll allow it ;-)

btw, you're welcome to mention it on our "Useful Programs" page:

- https://neomutt.org/contrib/useful-programs
- https://github.com/neomutt/neomutt.github.io/blob/main/_contrib/useful-programs.md

**Reply by pinecat** [reply]
@flatcap Seems like a fair trade :smile_cat: , you've been starred.

I'll see if I can put the PR together this weekend for the Useful Programs page. Thanks so much for your support!
